raster/src/main/scala/geotrellis/raster/io/geotiff/ArraySegmentBytes.scala

```diff
@@ -51,12 +51,9 @@ object ArraySegmentBytes {
             val result = Array.ofDim[Array[Byte]](offsets.size)
 
             cfor(0)(_ < offsets.size, _ + 1) { i =>
-              byteReader.position(offsets(i).toInt)
-              result(i) = byteReader.getSignedByteArray(byteCounts(i).toInt)
+              result(i) = byteReader.getSignedByteArray(byteCounts(i), offsets(i))
```

Comment:
`oldOffset` is unused. Looks like the previous version of this file used it to return position, I would guess thats not a good idea and this variable needs to be removed now.